pyiron_atomistics/atomistics/structure/atoms.py

```diff
@@ -1977,7 +1967,15 @@ def extend(self, other):
             )
             other = ase_to_pyiron(other)
 
-        new_indices = other.indices.copy()
+        d = self._store_elements.copy()
+        d.update(other._store_elements.copy())
+        chem, new_indices = np.unique(
+            self.get_chemical_symbols().tolist()
+            + other.get_chemical_symbols().tolist(),
+            return_inverse=True,
+        )
+        new_species = [d[c] for c in chem]
```

Comment:
It'd be nice to move this whole block into the if clause below where it is actually used.  Otherwise it just takes a while to figure out where it is needed.

Reply:
+1

Reply:
Actually it cannot be done, because ASE-`extend` has to be called after the species are checked. I presume that `indices` will be removed in the near future anyway, so I suggest to keep this location for now.